File: src/wesoga/blocks/BlockModel.java
package wesoga.blocks;

import wesoga.textures.Textures;

public final class BlockModel {
	public final boolean visible;

	public final int northTexture;
	public final int southTexture;
	public final int eastTexture;
	public final int westTexture;
	public final int topTexture;
	public final int botTexture;

	public BlockModel() {
		visible = false;

		northTexture = -1;
		southTexture = -1;
		eastTexture = -1;
		westTexture = -1;
		topTexture = -1;
		botTexture = -1;
	}

	public BlockModel(byte[] texture) {
		this.visible = true;

		int id = Textures.textureID(texture);

		northTexture = id;
		southTexture = id;
		eastTexture = id;
		westTexture = id;
		topTexture = id;
		botTexture = id;
	}

	@Override
	public final boolean equals(Object obj) {
		return this == obj;
	}
}

File: src/resources/web/block_model.ts
class BlockModel {
    constructor(visible: boolean, northTexture: number, southTexture: number, eastTexture: number, westTexture: number, topTexture: number, botTexture: number) {
        this.visible = visible;
        this.northTexture = northTexture;
        this.southTexture = southTexture;
        this.eastTexture = eastTexture;
        this.westTexture = westTexture;
        this.topTexture = topTexture;
        this.botTexture = botTexture;
    }

    public visible: boolean;

    public northTexture: number;
    public southTexture: number;
    public eastTexture: number;
    public westTexture: number;
    public topTexture: number;
    public botTexture: number;
}
